### PR TITLE
Updated to the Latest API

### PR DESCRIPTION
**Pull request by GabrielNotman**
This PR mainly updates HiberCalls.cpp to add support for the latest API. 

The function 'sendNMEA()' has been removed as it is no longer supported.
The function 'prepareBroadcast()' has been renamed as 'setPayload()'.

A few other fixes to with regards to how the responses are handled.

### CODE REVIEW

File: developer-kit/developer-kit.ino [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // This is an example project.
 
+#if defined(ARDUINO_ARCH_AVR)
 #include <avr/wdt.h>
+#endif
```
Comment (diamondo25) [inline developer-kit/developer-kit.ino:5]: Shouldnt this also remove watchdog timer stuff if its not supported?

File: developer-kit/Hiber.h [hunk 2]
```diff
@@ -1,4 +1,5 @@
-#pragma once
+#ifndef HIBER_H_
+#define HIBER_H_
```
Comment (diamondo25) [inline developer-kit/Hiber.h:2]: No more pragma once?

**Comment by diamondo25** [comment]
I didnt know about the removal of run_nmea. We need it anyway (the code behind it) in order to support gps chips

File: developer-kit/developer-kit.ino (hunk 1 above)
Comment (GabrielNotman) [reply developer-kit/developer-kit.ino:5]: That include is AVR specific, and I was building on SAMD based board.

There actually no calls to any of the watch dog timer functions anywhere in this code, so the include can be removed altogether.

File: developer-kit/Hiber.h (hunk 2 above)
Comment (GabrielNotman) [reply developer-kit/Hiber.h:2]: This is more of a style choice as pragma once is non-standard but widely supported.

**Comment by GabrielNotman** [comment]
With regards to the run_nmea() command. I'm not sure if this had anything to do with GPS NMEA strings, but I don't actually know what it was for (its wasn't in the old API documentation).

All the (removed) function did was send "run_nmea(NMEA_STRING)" to the modem and return whether the response received matched the ok response.

**Comment by mhogerheijde** [comment]
Closing PR, archiving repo

**Pull request closed without merging**